src/compute-client/src/controller/instance.rs

```diff
@@ -1693,6 +1667,9 @@ where
     }
 
     fn handle_response(&mut self, response: ComputeResponse<T>, replica_id: ReplicaId) {
+        if let Some(replica) = self.replicas.get(&replica_id) {
+            replica.metrics.inner.response_queue_size.dec();
+        }
```

Comment:
This seems like a strange place to update a queue size metric, since the queue it relates to is not touched here.

Maybe this is a good time to migrate to a `mz_ore::channel::instrumented_unbounded_channel`. We'll need to replace the metric though, since that channel expects to count up two metrics, rather than increasing/decreasing a single one.

Reply:
TIL, I'll switch it over.